plugins-extra/query_enhancements/server/search/async/sql_async_facet.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export class SQLAsyncFacet {
+  constructor(private client: any) {
+    this.client = client;
+  }
+
+  private fetch = async (request: any, format: string, responseFormat: string) => {
+    const res = {
+      success: false,
+      data: {},
+    };
+    try {
+      const params = {
+        body: {
+          query: request.body.query,
+          datasource: 'mys3',
```

Comment:
we should only define this one spot. the dataframe.name can probably have this and pass here right?